app/controllers/results_controller.rb

```diff
@@ -138,12 +138,11 @@ def edit
 
   def run_tests
     grouping_id = params[:grouping_id]
-    # TODO: The submission id is set incorrectly as the grouping_id
     submission_id = Result.find(params[:id]).submission.id
 
     begin
       AutomatedTestsHelper.request_a_test_run(grouping_id,
-                                              'submission',
+                                              'request',
```

Comment:
I think we are talking about two different "run tests" buttons. There is one in the admin/ta view for a collected submission, and one in the student view, Automated testing tab.
The student button is the one enabled by the token mechanism, and from my understanding is a way for students to test their code prior to submission, if enabled.